string string_in
---
string string_out
